lib/imap.php

```diff
@@ -85,12 +85,19 @@ public function checkPassword($uid, $password) {
 					$groups[] = $pieces[1];
 		}
 
-		$protocol = ($this->sslmode === "ssl") ? "imaps" : "imap";
-		$url = "{$protocol}://{$this->mailbox}:{$this->port}";
 		$ch = curl_init();
-		if ($this->sslmode === 'tls') {
```

Comment:
maybe a simpler solution: how about just changing this line to include everything expect:
- explicit `ssl` (and also TRUE), 
- explicit `nossl`, `notls`, maybe "empty string" (and also FALSE)
so everything else (including `tls`) would be handled as STARTTLS...
what do you think?